poncho/src/poncho/library_network_code.py

```diff
@@ -123,8 +123,20 @@ def start_function(in_pipe_fd, thread_limit=1):
                         event = cloudpickle.load(f)
 
                     # output of execution should be dumped to outfile.
-                    with open('outfile', 'wb') as f:
-                        cloudpickle.dump(globals()[function_name](event), f)
+                    result = globals()[function_name](event)
+                    try:
+                        with open('outfile', 'wb') as f:
+                            cloudpickle.dump(result, f)
+                    except Exception:
+                        if os.path.exits('outfile'):
+                            os.remove('outfile')
+                        raise
```

Comment:
I'm not sure about raising an exception here, because this means if a function fails to produce an output file the library dies with it too.

Reply:
This is the direct method, and even without these changes, the library dies if function fails.